### ISSUE TITLE
stl.py not python-3.4 compatible

### ISSUE THREAD

**Issue by rudregues** [WoLpH/numpy-stl#9]
I've installed numpy-stl with pip3 for python-3.4 and pip for python-2.7

The following code works with python 2.7:
<code>from stl import mesh</code>

But with python-3.4 I get the following error:
```bash
Traceback (most recent call last):
File "<stdin>", line 1, in <module>
File "/usr/local/lib/python3.4/dist-packages/stl/mesh.py", line 1, in <module> from . import stl
File "/usr/local/lib/python3.4/dist-packages/stl/stl.py", line 49   
 except RuntimeError, (recoverable, e):
                    ^
SyntaxError: invalid syntax
```
Editing line 49 to <code>except RuntimeError (recoverable, e):</code> removes the error and put another one:
```bash
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.4/dist-packages/stl/mesh.py", line 1, in <module>
    from . import stl
  File "/usr/local/lib/python3.4/dist-packages/stl/stl.py", line 155
    except AssertionError, e:
                         ^
SyntaxError: invalid syntax
```

**Comment by WoLpH** [comment]
Python 3 is indeed not working yet, I'll add it to my todo but it's slightly more work than I had hoped :(

**Comment by mattngc** [comment]
I did a test with 2to3.
$> 2to3 --output-dir=numpy-stl-3 -W -n numpy-stl-1.4.2
This seems to go a long way towards the changes needed. After doing this I could use pip to install the package and import it. Interestingly, the utilities work in one direction:

This works fine:
stl2ascii mybinary.stl myascii.stl

This fails:
$> stl2bin myascii.stl mybin.stl
Traceback (most recent call last):
  File "/home/matt/anaconda3/bin/stl2bin", line 9, in <module>
    load_entry_point('numpy-stl==1.4.2', 'console_scripts', 'stl2bin')()
  File "/home/matt/Downloads/test23/numpy-stl-3/stl/main.py", line 80, in to_binary
    update_normals=not args.use_file_normals)
  File "/home/matt/Downloads/test23/numpy-stl-3/stl/stl.py", line 198, in save
    write(fh, name)
  File "/home/matt/Downloads/test23/numpy-stl-3/stl/stl.py", line 227, in _write_binary
    fh.write(struct.pack('@i', self.data.size))
TypeError: must be str, not bytes

Obviously, this must be to the changes in string and byte handling in python 3. Would love to help more, but I'm not an expert.

**Comment by WoLpH** [comment]
That's indeed the problem I ran into, it's not impossible to fix but it's just taking slightly more time than I have right now :)

Still a work in progress though.

**Comment by mattngc** [comment]
ok, if I end up doing any more work on it for my application, I will try to post my findings here.

**Comment by WoLpH** [comment]
I've just committed a few changes to make it slightly better, but something is fundamentally different in how py3k handles things, just try and run the test suite and you'll see that something is a bit weird :)

**Comment by WoLpH** [comment]
I've just released version 1.5 which has Python 3 support as well as Python 2 support :)
